processing/src/main/java/org/apache/druid/java/util/emitter/service/SegmentMetadataEvent.java

```diff
@@ -85,7 +86,7 @@ public SegmentMetadataEvent(
   )
   {
     this.dataSource = dataSource;
-    this.createdTime = createdTime;
+    this.createdTime = createdTime != null ? createdTime : DateTimes.nowUtc();
```

Comment:
Nit: I don't think this field is ever passed in the current usages of this class.

Reply:
Yes. It is never passed as null right now. The current and only caller of the SegmentMetadataEvent.create pass DateTimes.nowUtc() but I want to be extra safe. This is in case some new user comes along and call  SegmentMetadataEvent.create without setting the createdTime (either though using the SegmentMetadataEvent.create without passing eventTime or calling the SegmentMetadataEvent constructor without setting createdTime.

Reply:
I agree, it makes sense to ensure that the `timestamp` is always non-null.
But if it is important for the field to be non-null, then we should be more explicit about the value we use for it. We should probably avoid silently converting a null to "now" (though there are other places in the Druid code that do it 😅 ).

So I would advise to do the following:
- Retain the public constructor but do a validation of this field thus forcing callers to pass a non-null value.
- Remove arg `eventTime` from method `.create()` and always use `DateTime.nowUtc()`.
- Add a javadoc to both the constructor and `create()` clarifying this point.

Reply:
I am doing something similar to what `ServiceMetricEvent` does, which most other metrics use. This is what @gianm suggested at https://apachedruidworkspace.slack.com/archives/C0309C9L90D/p1715638010942329?thread_ts=1715634244.240639&cid=C0309C9L90D

Reply:
Fair enough, this comment was not a blocker anyway. We can proceed with this merge and cleanup usage of both `ServiceMetricEvent` and `SegmentMetadataEvent` later.